src/python/pants/backend/python/util_rules/extract_pex_test.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import Iterable
+
+import pytest
+
+from pants.backend.python.util_rules import extract_pex, pex
+from pants.backend.python.util_rules.extract_pex import ExtractedPexDistributions
+from pants.backend.python.util_rules.pex import (
+    Pex,
+    PexInterpreterConstraints,
+    PexRequest,
+    PexRequirements,
+)
+from pants.core.util_rules.pants_environment import PantsEnvironment
+from pants.engine.fs import EMPTY_DIGEST, Snapshot
+from pants.testutil.option_util import create_options_bootstrapper
+from pants.testutil.rule_runner import QueryRule, RuleRunner
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=[
+            *extract_pex.rules(),
+            *pex.rules(),
+            QueryRule(Pex, [PexRequest, PantsEnvironment]),
+            QueryRule(ExtractedPexDistributions, [Pex]),
+        ]
+    )
+
+
+def get_distributions(
+    rule_runner: RuleRunner, *, requirements: Iterable[str], constraints: Iterable[str]
+) -> ExtractedPexDistributions:
+    # NB: The constraints are important for determinism.
+    rule_runner.create_file("constraints.txt", "\n".join(constraints))
+
+    pex_request = PexRequest(
+        output_filename="test.pex",
+        requirements=PexRequirements(requirements),
+        interpreter_constraints=PexInterpreterConstraints([">=3.6"]),
+        internal_only=True,
+    )
+    options_bootstrapper = create_options_bootstrapper(
+        args=[
+            "--backend-packages=pants.backend.python",
+            "--python-setup-requirement-constraints=constraints.txt",
+        ]
+    )
+    built_pex = rule_runner.request(Pex, [pex_request, options_bootstrapper, PantsEnvironment()])
+    return rule_runner.request(ExtractedPexDistributions, [built_pex])
+
+
+def test_extract_distributions(rule_runner: RuleRunner) -> None:
+    constraints = ["six==1.15.0", "t61codec==1.0.1", "x690==0.2.0"]
```

Comment:
I believe there was a specific reason to choose `x690`? Could we add a quick comment about that?

Reply:
For this test, `x690` in particular isn't important (it is for MyPy). But it is important that none of these have wheels specific to the interpreter, like `cp36` vs. `cp37`. I'll comment about that.